src/Cli/dotnet/commands/dotnet-test/Terminal/TerminalTestReporter.cs

```diff
@@ -801,6 +800,7 @@ private static void AppendExecutableSummary(ITerminal terminal, int? exitCode, s
 
     private static void AppendAssemblySummary(TestProgressState assemblyRun, ITerminal terminal)
```

Comment:
Just to make sure, are we sure no one else could be writing to the console concurrently during the execution of any method in this class?

Reply:
Yes that should be enforced by the batching in terminal logger.